docs/DeveloperGuide.md

```diff
@@ -1,38 +1,160 @@
 # Developer Guide
 
+## Table of Contents
+
+<!-- TOC -->
+* [Acknowledgements](#acknowledgements)
+* [Product Scope](#product-scope) 
+  * [Target User Profile](#target-user-profile)
+  * [Value Proposition](#value-proposition)
+* [Design & Implementation](#design--implementation)
+  * [Architecture](#architecture)
+  * [UI component](#ui-component)
+  * [Start component](#start-component)
+  * [Add component](#add-component)
+  * [List component](#list-component)
+  * [View component](#view-component)
+  * [Delete component](#delete-component)
+  * [Exit component](#exit-component)
+* [User Stories](#user-stories)
+  * [V1.0](#v10)
+  * [V2.0](#v20)
+* [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+* [Instruction for Manual Testing](#instructions-for-manual-testing)
+<!-- TOC -->
+
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+###### [Back to table of contents](#table-of-contents)
+
+## Product Scope
+
+### Target User Profile
+* People who go to gym and do exercise
+* People who want to lose weight or strength muscles
+
+### Value Proposition
+Help our target users to record down the exercise, and they can check the exercise whenever they want.
+Besides, this fitness app also help them to calculate the calories consumption for each workout.
 
-## Design & implementation
 
+###### [Back to table of contents](#table-of-contents)
+
+## Design & Implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+###### [Back to table of contents](#table-of-contents)
+
+We aim to help fellow gym enthusiasts and new gym members to help keep
+track of their future and present training.
+=======
+### UI component
+
+###### [Back to table of contents](#table-of-contents)
+
+### Start component
+
+###### [Back to table of contents](#table-of-contents)
+
+### Add component
+
+The add mechanism is facilitated by `AddCommand`. 
+It extends `Command` and modifies the execute function to add an exercise.
+
+<img src="images/AddExerciseDiagram.png" width="450" />
```

Comment:
Perhaps it would be helpful to add activation bars in the diagram to show the timeline of the methods being called